common/include/bmp180_protocol.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * baroceptor_protocol.h
+ *
+ *  Created on: 2020��10��19��
+ *      Author: duanning
+ */
+
+#ifndef BMP180_PROTOCOL_H_
+#define BMP180_PROTOCOL_H_
+
+#include "abstract_protocol.h"
+
+class Bmp180Protocol: public DummyProtocol {
+public:
+	Bmp180Protocol() {}
+	~Bmp180Protocol() {}
+
+	virtual error_condition UnpackFrame(
+			const vector<uint8_t>& buffer, vector<uint8_t>& unpacked_data) {
+		if (ExtractValidData(buffer, unpacked_data) != true) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+private:
+	virtual error_condition CalcCheckCode(\
+			const vector<uint8_t>& buffer, vector<uint8_t> check_code) {
+		if (buffer.back() != check_code.back()) {
+			return make_error_condition(errc::bad_message);
+		}
+		if (buffer.at(buffer.size() - 2) !=\
+				check_code.at(check_code.size() - 2)) {
+			return make_error_condition(errc::bad_message);
+		}
+		return kNoError;
+	};
+	virtual bool ExtractValidData(const vector<uint8_t>& buffer,\
+			vector<uint8_t>& unpacked_data) {
+		if (IsFrameComplete(buffer) != true) {
+			return false;
+		}
+		for (auto i : buffer)
+			cout << hex << int(i) << " ";
+		cout <<endl;
+
+		if (CalcCheckCode(buffer, terminator) != kNoError) {
+			return false;
+		}
+		unpacked_data.assign(buffer.begin(), buffer.begin() + buffer.size() -\
+				terminator.size());
+
+		return true;
+	};
+
+	vector<uint8_t> terminator = {0x01, 0x0A};
```

Comment:
This one is supposed to be private and we need to change the name, adding a `_` at the end.